libs/filabridge/src/SamplerBindingMap.cpp

```diff
@@ -43,14 +44,23 @@ void SamplerBindingMap::populate(const SamplerInterfaceBlock* perMaterialSib,
         if (sib) {
             auto sibFields = sib->getSamplerInfoList();
             for (const auto& sInfo : sibFields) {
-                if (offset > maxSamplerIndex) {
+                if (numSampler > maxSamplerIndex) {
```

Comment:
This check needs to be moved out of the  outer `for` loop. Better yet, remove the `overflow` variable and check `if (numSampler > maxSamplerIndex)` on line 70.

Reply:
Thanks. It also has a bug. I make a lambda function to check overflow.